packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -149,8 +154,9 @@ export class DebugConfigurationManager {
         this.updateCurrent(option);
     }
     protected updateCurrent(options: DebugSessionOptions | undefined = this._currentOptions): void {
-        this._currentOptions = options
-            && this.find(options.configuration.name, options.workspaceFolderUri);
+        const isDynamic = options?.configuration.dynamic;
+        this._currentOptions = options && !isDynamic ? this.find(options.configuration.name, options.workspaceFolderUri) : options;
```

Comment:
I'll leave the final decision up to you, the following can be updated if you'd like:

```suggestion
        this._currentOptions = options && !options.configuration.dynamic ? this.find(options.configuration.name, options.workspaceFolderUri) : options;
```